EssentialApp/EssentialAppTests/CommentsUIIntegrationTests.swift

```diff
@@ -0,0 +1,267 @@
+		}
+		return value
+	}
+	
+	private func assert(_ sut: CommentsController, isRendering comments: [Comment], file: StaticString = #file, line: UInt = #line) {
+		guard sut.numberOfRenderedCommentViews == comments.count else {
+			return XCTFail("Expected \(comments.count) comments, got \(sut.numberOfRenderedCommentViews) instead", file: file, line: line)
+		}
+		comments.enumerated().forEach { index, comment in
+			assertThat(sut, hasViewConfiguredFor: comment, at: index, file: file, line: line)
+		}
+	}
+	
+	private func assertThat(_ sut: CommentsController, hasViewConfiguredFor comment: Comment, at index: Int, file: StaticString = #file, line: UInt = #line) {
+		let view = sut.commentView(at: index)
+		
+		guard let cell = view as? CommentCell else {
+			return XCTFail("Expected \(CommentCell.self) instance, got \(String(describing: view)) instead", file: file, line: line)
+		}
+		
+		let formatter = RelativeDateTimeFormatter()
+		let relativeDate = formatter.localizedString(for: comment.createdAt, relativeTo: Date())
+		
+		XCTAssertEqual(cell.author, comment.author, "author at index \(index)", file: file, line: line)
+		XCTAssertEqual(cell.date, relativeDate, "date at index \(index)", file: file, line: line)
+		XCTAssertEqual(cell.comment, comment.message, "message at index \(index)", file: file, line: line)
+	}
+	
+	private func makeComment(id: UUID = UUID(), message: String = "message", createdAt: Date = Date(), author: String = "author") -> Comment {
+		return Comment(id: id, message: message, createdAt: createdAt, author: author)
+	}
+	
+	class LoaderSpy: CommentsLoader {
+		private var completions = [(CommentsLoader.Result) -> Void]()
+		var loadCallCount: Int {
+			return completions.count
+		}
+		
+		var cancelCallCount = 0
+		
+		private final class CancelableTaskSpy: CancelableTask {
+			private let onCancel: () -> Void
+			
+			init (onCancel: @escaping () -> Void) {
+				self.onCancel = onCancel
+			}
+			
+			func cancel() {
+				onCancel()
+			}
+		}
+		
+		func load(completion: @escaping (CommentsLoader.Result) -> Void) -> CancelableTask {
+			completions.append(completion)
+			return CancelableTaskSpy() { [weak self] in
+				self?.cancelCallCount += 1
+			}
+		}
+		
+		func completeCommentsLoading(with comments: [Comment] = [], at: Int) {
+			completions[at](.success(comments))
+		}
+		
+		func completeCommentsLoadingWithError(at: Int) {
+			let error = NSError(domain: "an error", code: 0)
+			completions[at](.failure(error))
+		}
+	}
+}
+
+private extension CommentsController {
+	func simulateUserInitiatedCommentsReload() {
+		refreshControl?.simulatePullToRefresh()
+	}
+	
+	func simulateUserTapOnErrorMessage() {
+		errorView?.simulateTap()
+	}
+	
+	var isShowingLoadingIndicator: Bool {
+		return refreshControl?.isRefreshing == true
+	}
+	
+	var numberOfRenderedCommentViews: Int {
+		return tableView.numberOfRows(inSection: 0)
+	}
```

Comment:
Create a computed var `commentsSection` to hide the value "0" of the section. So if the section ever changes, you just need to update the helper instead of updating everywhere that references the value "0".

For example:

```suggestion
	var numberOfRenderedCommentViews: Int {
		return tableView.numberOfRows(inSection: commentsSection)
	}
	
	var commentsSection: Int { 0 }
```